swift/internal/linking.bzl

```
@@ -236,14 +236,18 @@ def register_link_binary_action(
                 "-framework",
                 objc.dynamic_framework_names.to_list(),
             ))
-            dep_link_flags.extend(static_framework_files)
+            dep_link_flags.extend([
+                file.path
+                for file in static_framework_files
+            ])
 
             linking_contexts.append(
                 cc_common.create_linking_context(
                     linker_inputs = depset([
                         cc_common.create_linker_input(
                             owner = owner,
                             user_link_flags = depset(dep_link_flags),
+                            additional_inputs = depset(static_framework_files),
```

We could probably use the original depset here, instead of recreating another.

```suggestion
                            additional_inputs = objc.static_framework_file,
```

---

Yea I felt weird about pulling from 2 places in either direction, but I'll go with this